ra_aid/file_listing.py

```
@@ -95,42 +99,66 @@ def get_file_listing(
+
+        # Apply limit if specified
         if limit is not None:
-            files = files[:limit]
+            all_files = all_files[:limit]
 
-        return files, total_count
+        return all_files, total_count
 
-    except subprocess.CalledProcessError as e:
-        raise GitCommandError(f"Git command failed: {e}")
+    except (DirectoryNotFoundError, DirectoryAccessError, GitCommandError) as e:
```

Ruff (0.8.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)